tsdb/engine/tsm1/encoding.go

```diff
@@ -209,27 +209,26 @@ func decodeFloatBlock(block []byte) ([]Value, error) {
 	dec := NewTimeDecoder(tb)
 	iter, err := NewFloatDecoder(vb)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	// Decode both a timestamp and value
-	var a []Value
 	for dec.Next() && iter.Next() {
 		ts := dec.Read()
 		v := iter.Values()
-		a = append(a, &FloatValue{ts, v})
+		*a = append(*a, &FloatValue{ts, v})
```

Comment:
I'm missing something.  How does not just keep growing continually in size since it's being reused?  Can explain?